### PR TITLE
CDK Destinations: DB Metadata calls cleanup

### PR DESCRIPTION
**Pull request by gisripa**
## What
* Cleaning up unsafe metadata calls

### CODE REVIEW

**Comment by gisripa** [comment]
* `master`
  * **#34533** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/34533?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gisripa and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/db/jdbc/DefaultJdbcDatabase.java [hunk 1]
```diff
@@ -73,6 +73,17 @@ public <T> Stream<T> unsafeResultSetQuery(final CheckedFunction<Connection, Resu
         });
   }
 
+  /**
+   * Implementations of DatabaseMetadata hold a reference of the Connection object. It is safe to use this to retrieve
```
Comment (edgao) [inline airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/db/jdbc/DefaultJdbcDatabase.java:77]: maybe worth adding a `DANGER!!!` or something, to make this more obvious

idk if it's correct to deprecate though, since e.g. getIdentifierQuoteString is a totally valid usecase :/

**Review by edgao — APPROVED** [review]
:heart: the comment, super informative

File: airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/db/jdbc/DefaultJdbcDatabase.java [hunk 2]
```diff
@@ -73,6 +74,17 @@ public <T> Stream<T> unsafeResultSetQuery(final CheckedFunction<Connection, Resu
...
+   * static information like getIndentifierQuoteString() etc but calling methods which return a ResultSet needs the connection
+   * to be still open. This may or may not work depending on how the underlying Connection object is handled
+   * eg. Hikari's ProxyConnection is not actually closed, rather recycled into Pool.
+   * See {@link #execute(CheckedConsumer)} which gives the caller a safe alternative to access ResultSet methods of DatabaseMetadata
```
Comment (edgao) [inline airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/db/jdbc/DefaultJdbcDatabase.java:82]:
```suggestion
   * See {@link #executeMetadataQuery(Function)} which gives the caller a safe alternative to access ResultSet methods of DatabaseMetadata
```

File: airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/db/jdbc/DefaultJdbcDatabase.java [hunk 3]
```diff
@@ -73,6 +74,17 @@ public <T> Stream<T> unsafeResultSetQuery(final CheckedFunction<Connection, Resu
         });
   }
 
+  /**
+   * Implementations of DatabaseMetadata hold a reference of the Connection object. It is safe to use this to retrieve
```
Comment (edgao) [inline airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/db/jdbc/DefaultJdbcDatabase.java:78]: I think this comment only needs to exist on the interface, this implementation class should just inherit it?

**Review by edgao — APPROVED** [review]
nitpicky comments about comments, otherwise lgtm

File: airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/db/jdbc/DefaultJdbcDatabase.java (hunk 3 above)
Comment (gisripa) [reply airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/db/jdbc/DefaultJdbcDatabase.java:78]: oh right. i wrote the impl first before interface 😅

File: airbyte-cdk/java/airbyte-cdk/airbyte-commons/src/main/java/io/airbyte/commons/concurrency/CompletableFutures.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+  /**
+   * Utility method which blocks until all futures are complete. Returns a list of the results of all
+   * futures.
+   *
+   * @param futures
+   * @return
+   * @param <T>
+   */
+  public static <T> List<T> allOf(final List<CompletableFuture<T>> futures) {
```
Comment (edgao) [inline airbyte-cdk/java/airbyte-cdk/airbyte-commons/src/main/java/io/airbyte/commons/concurrency/CompletableFutures.java:20]: is this file needed? not in use afaict, and the implementation looks broken

File: airbyte-cdk/java/airbyte-cdk/airbyte-commons/src/main/java/io/airbyte/commons/json/Jsons.java [hunk 5]
```diff
@@ -141,6 +141,20 @@ public static Optional<JsonNode> tryDeserialize(final String jsonString) {
     }
   }
 
+  /**
+   * This method does not generate deserialization warn log on why serialization failed.
```
Comment (edgao) [inline airbyte-cdk/java/airbyte-cdk/airbyte-commons/src/main/java/io/airbyte/commons/json/Jsons.java:145]:
```suggestion
   * This method does not generate deserialization warn log on why serialization failed.
   * See also {@link #tryDeserialize(String)}.
```

**Review by edgao — APPROVED** [review]

File: airbyte-cdk/java/airbyte-cdk/airbyte-commons/src/main/java/io/airbyte/commons/concurrency/CompletableFutures.java (hunk 4 above)
Comment (gisripa) [reply airbyte-cdk/java/airbyte-cdk/airbyte-commons/src/main/java/io/airbyte/commons/concurrency/CompletableFutures.java:20]: Created for something else but removed the impl for now on this branch. I'll remove it later if not needed.
